File: day01/challenge2.js
const helpers = require('../helpers');

const input = helpers.readlines(__dirname + '/input').map(v => parseInt(v))

for(let i = 0; i < input.length; i++) {
  let a = input[i];
  for (let j = 0; j < input.length; j++) {
    let b = input[j];
    for (let k = 0; k < input.length; k++) {
      let c = input[k];
      let sum = a + b + c;
      if ( sum === 2020 ) {
        console.log(a * b * c);
        process.exit(); // assumes a single match
      }
    }
  }
}

File: day01/challenge2.rb
input = File.readlines(__dir__ + '/input').map { |l| l.gsub(/\n/, '').to_i}

input.each do |i|
  input.each do |j|
    input.each do |k|
      sum = i + j + k
      if sum == 2020
        puts i * j * k
        exit
      end
    end
  end
end